Custom Ticks

thejeff77 opened the pull request:
Added custom ticks to the HUMSlider. 

Added various option parameters to enhance the customization potential for the slider.

Commit:
Adding methods for dynamic tick marks

Commit:
Tick object, tick layout constraints.

Commit:
Cleanup of unused code.

Finalizing custom tick spacing and animation setup.

Add var to toggle tick transparency.

Commit:
Trying another approach.

Commit:
tick spacing is working

Commit:
Final cleanup

Commit:
Updating the readme

Commit:
fixing heading formatting

Commit:
Added comment to clarify refreshView param

Commit:
Resolving minor code comments

thejeff77 committed:
Update project.pbxproj

Removing team info from PR

Commit:
Changing to NSUInteger

Commit:
Merge branch 'master' of https://github.com/thejeff77/HUMSlider

Commit:
Updating defaults to documented values.

Commit:
oops

thejeff77 commented:
I've fixed the track width issue we talked about. Currently passing in a Tick with a 0 value will inset the tick from the left enough so that it is in the center of the thumb image, which is about 15px from left by default. I was wondering if I should provide an option to disable inlay of the tick by the radius of the thumb image, and decided to be agile and maybe wait until someone asked for that. This is working fantastically now. It would probably be super easy to modify the previous code to use the new shorter layout methods for odd or evenly spaced ticks. I'll be opening later PRs to add a color to the tick itself so that the default color can be overridden on a tick by tick basis.

Commit:
reverting default size changes

Commit:
added logic to squeeze the ticks to the slider

Commit:
Made the ticks animate and squeeze to the slider.
Fixed a bug where the custom thumb adjustment value being positive should pop the tick up higher instead of the inverse, because the differentials are more negative the higher they pop up.

Commit:
Added a remove all method

Commit:
Added App Icon full view.

Commit:
fixing main thread warnings

Commit:
Removed unnecessary methods

thejeff77 commented on Library/HUMSlider.m:
```diff
@@ -15,15 +15,15 @@
 
 // Default Constants
-static CGFloat const DefaultThumbPxWidth = 30; //Size of apple's default thumb icon.
+static CGFloat const DefaultThumbPxWidth = 31; //Size of apple's default thumb icon.
```
I believe this can be removed

Commit:
Fixed bugs switching between the two slider types

designatednerd commented:
@thejeff77 Mind if I reopen this? Was planning on getting you more feedback this weekend or next week

Commit:
Removed copyright as it interferes with the MIT license. Added comments and changed an integer reference.

thejeff77 commented:
No problem, I have a tricky bug to fix, and realized I have quite a bit of cleanup to do. I'm working on this more today, and can re-open as soon as the code looks a little better. I hope that works! I did a major refactor & am making sure all of the old and new functionality works together. I could never have done this without the code there in the first place to use as an example. Thanks :)

designatednerd commented:
Awesome! Ping me when it's up and I'll take a look

Commit:
Cleanup and bugfixing

Commit:
Touched up the readme

thejeff77 commented:
All done. Much better now. Pretty much re-factored how the consistently spaced tick marks worked - it no longer requires an odd number. It didn't make sense to have two separate implementations in there - one for custom ticks the other for consistently spaced ticks. Also added a feature that hugs the tick to the outside of the slider as it moves by - this is cool when the slider is moving slowly for playback type stuff. Added some public properties for more control over how the slider looks when dragged / not dragged. More specifically - one var to keep them from disappearing when the user isn't touching the slider. Another one to prevent them from being lowered while the slider isn't being dragged. Also removed the copyright, as this causes issues with open source review boards and can be perceived as a conflict with an open source MIT license.

Commit:
Final cleanup and merging

Commit:
Changing double to CGFloat

Commit:
using self. instead of synthesized

Commit:
use BOOL not bool

Commit:
Fixed index insertion error for middle constraints

Commit:
Removing global track Rect

Commit:
Removing unused var

thejeff77 commented:
@designatednerd this is ready for a review.

Commit:
Fixed issue where the tick doesn't pop or show on initial addition

Commit:
Real fix for a transparency issue
